src/main/java/org/jpeek/calculus/GeneralCalculation.java

```diff
@@ -0,0 +1,181 @@
+
+    /**
+     * Created report with aggregated value.
+     *
+     * @param target Path to report file
+     * @param metrics List of metrics used in calculation
+     * @throws IOException If fails
+     */
+    @SuppressWarnings({
+        "PMD.AvoidDuplicateLiterals",
+        "PMD.ExcessiveMethodLength"
+    })
+    public void createReport(final Path target, final List<String> metrics) throws IOException {
+        final Map<String, MetricPresentation> res = new HashMap<>();
+        for (final String metric : metrics) {
+            res.put(
+                metric,
+                new MetricPresentation(
+                    String.format(
+                        "%s/%s.html",
+                        target.toString(),
+                        metric
+                    )
+                )
+            );
+        }
+        final Map<String, String[]> output = getResult(res);
+        String template;
+        template = IOUtils.toString(
+            GeneralCalculation.class.getResourceAsStream("general_calculus.html")
+        );
+        final String classesout = String.format(
+            "<p id=\"classes\">Classes: %s.</p>\n",
+            output.keySet().size()
+        );
+        String tbody = "";
+        for (final Map.Entry<String, String[]> entry : output.entrySet()) {
+            tbody = tbody.concat(
+                String.format(
+                    "%s%s%s%s%s%s", "<tr>\n",
+                    String.format(
+                        "               <td><code title=\"%s\">%s</code></td>\n",
+                        entry.getKey(),
+                        entry.getKey()
+                    ),
+                    String.format(
+                        "               <td style=\"text-align:right;\">%s</td>\n",
+                        entry.getValue()[0]
+                    ),
+                    String.format(
+                        "               <td style=\"text-align:right\">%s</td>\n",
+                        entry.getValue()[1]
+                    ),
+                    String.format(
+                        "               <td style=\"text-align:right\">%s</td>\n",
+                        entry.getValue()[2]
+                    ),
+                    "            </tr>"
+                )
+            );
+        }
+        template = template.replace(
+            "<p id=\"classes\"></p>",
+            classesout
+        );
+        template = template.replace(
+            "<tbody id=\"table_body\"></tbody>",
+                String.format(
+                    "<tbody id=\"table_body\">\n%s\n</tbody>",
+                    tbody
+                )
+            );
+        try {
+            final File file = new File(String.format("%s%s", target.toString(), "/total.html"));
+            final BufferedWriter bwriter = Files.newBufferedWriter(file.getAbsoluteFile().toPath());
+            bwriter.write(template);
+            bwriter.close();
+        } catch (final IOException ioex) {
```

Comment:
@iMaks99 I think we have a more general mechanism in the project for exception handling. Here, I suggest, we just let the exception float up.